[TEST-FAILURE] QueryIndexingTest.testResultsHaveNullFields_whenUsingIndexes

```
java.lang.AssertionError: expected:<2500> but was:<2429>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:743)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:555)
	at org.junit.Assert.assertEquals(Assert.java:542)
	at com.hazelcast.map.query.QueryIndexingTest.testResultsHaveNullFields_whenUsingIndexes(QueryIndexingTest.java:92)
```
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance-OpenJDK8/com.hazelcast$hazelcast/59/testReport/junit/com.hazelcast.map.query/QueryIndexingTest/testResultsHaveNullFields_whenUsingIndexes/

--- a/hazelcast/src/test/java/com/hazelcast/map/query/QueryIndexingTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/map/query/QueryIndexingTest.java
@@ -54,7 +54,6 @@ public class QueryIndexingTest extends HazelcastTestSupport {
     @Before
     public void waitForCluster() {
         assertClusterSizeEventually(2, h1);
-        waitClusterForSafeState(h1);
     }
 
     @Test
@@ -62,6 +61,7 @@ public class QueryIndexingTest extends HazelcastTestSupport {
 
         IMap<Integer, Employee> imap1 = h1.getMap("employees");
         imap1.putAll(employees);
+        waitAllForSafeState();
 
         Collection<Employee> matchingEntries = runQueryNTimes(3, h2.<String, Employee>getMap("employees"));
 
@@ -80,6 +80,7 @@ public class QueryIndexingTest extends HazelcastTestSupport {
         imap1.addIndex("city", true);
 
         imap1.putAll(employees);
+        waitAllForSafeState();
 
         Collection<Employee> matchingEntries = runQueryNTimes(3, h2.<String, Employee>getMap("employees"));
         assertEquals(count/2, matchingEntries.size());

